### DISCUSSION TITLE
Session string error

### DISCUSSION THREAD

**Discussion by Digi777** [aahnik/tgcf#460] [Q&A] [answered]
It looks like the session string tool https://replit.com/@aahnik/tg-login?v=1 is not working anymore, i can add the API ID, API hash, and then after it tells me to press [Enter] nothing happens

Any idea how to fix this?

**Comment by cold-nagai** [comment]
Run tg-login locally :)

**Comment by Digi777** [comment]
Do you guys have a tutorial on how to do this?

**Reply by RafleeA** [reply]
You meant Run tg-login locally on your windows machine?
Use Visual Studio mate (make sure you had latest Python installed / 3.10 above)
Simply Install :
`pip install tg-login`
Run : 
`tg-login`

**Reply by Digi777** [reply]
Mmh sorry for my noobness, but after using pip install tg-login in the chosen folder I can't run the program.

And when I check the folder it appears as empty...

Could you maybe explain me how to run it? I'm an absolute beginner and not dev (obviously) but I'm interested to learn. I know that all this is very easy but I can't get it work...

**Reply by Digi777** [reply]
If i write `pip show tg-login` I can see the program. But after this step I'm totally lost. I have no idea what to do. 

If I write `import tg-login` in the command line or in a python file I get an error message telling me that there isn't such module there

**Reply by iamyashg** [reply]
that too wont work as there is issue with telethon port and that can fix by editing it in replit by viewing hidden files

but i will suggest you easy method for creating string session : use this telegram bot https://t.me/SessionStringGeneratorZBot

**Reply by ghost** [reply, marked as answer]
I assume you use Windows. Go to Desktop and create a new folder, from inside the folder open a console and run `python -m venv .venv`. Then run `.venv/scripts/activate`, then `pip install tg-login`. Wait to finish, then run `tg-login`. After you give all things required will give you session string in Saved Messages.

**Reply by Digi777** [reply]
@poliandro it worked, thanks!

**Comment by MobCode100** [comment]
On Linux, running tg-login locally works just fine. You don't need to install tg-login as it is provided alongside with tgcf. Any error messages for this problem? 

For windows users, one idea is to install WSL, and install tgcf inside it. Download WSL from windows store, and install. Once on command prompt or powershell, just type in `wsl` and enter. Then, proceed to install tgcf normally and after that run tg-login.

**Reply by Digi777** [reply]
Hey, thanks for the reply, it looks like installing Linux was not needed, I finally made it work on windows directly. Thanks for your help anyway!

**Reply by MobCode100** [reply]
Happy for you!
